Source/MLX/MLXArray+Init.swift

```diff
@@ -160,8 +177,10 @@ extension MLXArray {
                 #endif
                 case .float32:
                     self.init(Float32(v))
-                case .bfloat16, .complex64:
-                    fatalError("dtype \(dtype) not supported")
```

Comment:
It was unclear at the time how these might be needed and I didn't have the right support to create them.